cosmwasm/submit-proposal.js

```diff
@@ -161,6 +161,41 @@ const execute = async (client, wallet, config, options) => {
     await callSubmitProposal(client, wallet, config, options, proposal);
 };
 
+const registerItsChain = async (client, wallet, config, options) => {
+    const chains = options.chains.map((chain) => {
+        const chainConfig = getChainConfig(config, chain);
+
+        if (!chainConfig.maxUintBits) {
+            throw new Error(`Missing maxUintBits for chain ${chain}`);
+        }
+
+        if (!chainConfig.maxDecimalsWhenTruncating) {
+            throw new Error(`Missing maxDecimalsWhenTruncating for chain ${chain}`);
+        }
+
+        const itsEdgeContract = chainConfig.contracts.InterchainTokenService?.address || chainConfig.contracts.ITS?.objects?.ChannelId;
```

Comment:
Create a helper called `getContractIdentifier(chain, contractName)` that hides this detail for now (it can check if chainType is `sui` and use the channelId instead)

Reply:
moved to helper function, however I didn't use the chainType for sui because it seems `chainType` is not being set for sui chains. Already talked to Joao about it